src/lib/registry.js

```diff
@@ -87,3 +90,23 @@ export function registerWidgetValueSerializer(widgetName, serializer) {
 export function getWidgetValueSerializer(widgetName) {
   return registry.widgetValueSerializers[widgetName];
 };
+
+/**
+ * Backend API
+ */
+export function registerBackends(name, BackendClass) {
+  if (!name || !BackendClass) {
+    console.error("Backend parameters invalid. example: CMS.registerBackends('myBackend', BackendClass)");
+  } else if (registry.backends[name]) {
+      console.error(`Backend [${ name }] already registered. Please choose a different name.`);
+  } else {
+    registry.backends[name] = {
+      init: config => new BackendClass(config),
```

Comment:
I still wonder why we shouldn't just do
```js
registry.backends[name] = BackendClass;
```
and then in the backend code
```js
return new Backend(new (getBackend(name))(config), name, authStore);
```

Can you explain? Just cleaner?

Reply:
coder preference, I like the object reference is all

Reply:
Thanks for taking initiative on this @talves, I love how simple of a first step this is!

Regarding this piece of code, we want project style to trump an individual developer's preference - no reason to make this function different than the others. Let's update this registry function to simply assign the registered entity as Caleb suggested.

After that, should be good to merge.

Reply:
I don't agree.

Reply:
I'm fine either way, it's pretty much just bikeshedding IMO. Do we have a reason that it needs to be one way or another?

Reply:
@talves Did you have a specific use case you envisioned here, like eventually making it work with regular objects, instead of class backends?

Reply:
It'd be different if we weren't committing to this decision as a part of our public API, which we can't break until 2.0 once merged. We should be in agreement before proceeding - let's feature flag it for now and get some mileage on it before making it public.

Reply:
Like Tony said, this specific discussion is just about code style -- it doesn't impact the public API at all.

Reply:
On not breaking the public API, though, I'm fine with putting it in a beta/flag.